.github/workflows/test.yml

```diff
@@ -19,15 +23,45 @@ jobs:
         with:
           go-version: 1.19
       - uses: actions/checkout@v3
-      - run: ./hack/make engine:build
       - run: echo $PWD/bin >> $GITHUB_PATH
       - run: ./hack/make engine:test
+        env:
+          _EXPERIMENTAL_DAGGER_JOURNAL: "/tmp/journal.log"
+      - run: ./hack/make engine:build
       - name: "ALWAYS print engine logs - especialy useful on failure"
         if: always()
         run: docker logs dagger-engine.dev
       - name: "ALWAYS print kernel logs - especialy useful on failure"
         if: always()
         run: sudo dmesg
+      - run: go build ./cmd/otel-collector
+        if: always()
+      - run: |
+          ./otel-collector /tmp/journal.log \
+            --name "${{ github.repository }}#${{ github.ref }}@${{ github.run_number }}.${{ github.run_attempt }}" \
+            --label "github.repository=${{ github.repository }}" \
+            --label "github.ref=${{ github.ref }}" \
+            --label "github.workflow=${{ github.workflow }}" \
+            --label "github.job=engine" \
+            --label "github.run_number=${{ github.run_number }}" \
+            --label "github.run_attempt=${{ github.run_attempt }}" \
+            > ./comment.md
+        shell: bash
+        if: always()
+        env:
+          OTEL_SERVICE_NAME: "dagger"
+          OTEL_EXPORTER_OTLP_ENDPOINT: "https://tempo-us-central1.grafana.net:443"
+          OTEL_EXPORTER_OTLP_HEADERS: ${{ secrets.OTEL_EXPORTER_OTLP_HEADERS }}
+          GRAFANA_CLOUD_USER_ID: "356840"
+          GRAFANA_CLOUD_URL: "https://logs-prod-017.grafana.net"
+          GRAFANA_CLOUD_API_KEY: ${{ secrets.GRAFANA_CLOUD_API_KEY }}
```

Comment:
Good point! Forgot about this.

@gerhard I believe you mentioned you could take care of this part? I don't remember what the solution was to make this work on PRs.

Reply:
Yes. I am re-adding it now as a regular environment variable. Since we scope it to `Write: logs` only, that is the only permission that this API key will have. No need to keep it a secret.

Reply:
@aluzzardi just pushed, please pull before adding more changes: https://github.com/dagger/dagger/pull/4340/commits/59ba4ba6fd3ba0ee69aef23cc089ca225c66a0df

Reply:
Will resolve this conversation as soon as I confirm that this works as expected. I will also clean up `secrets.GRAFANA_CLOUD_API_KEY` from GitHub Actions.

Reply:
Nice -- also need to do the same for the OTEL headers for Tempo

Reply:
All done.